### ISSUE TITLE
Custom ranking?

### ISSUE THREAD

**Issue by ahadda5** [terrierteam/pyterrier_t5#8]
Can I use your package on a custom dataset?  I need to rank 25 targets/labels for corpus. 
I read that it only supports pre-trained models?   I want to fine-tune a model(any T5 size) on a a custom dataset for the task of ranking. 

thanks

**Comment by cmacdonald** [comment]
Of course, monoT5 takes a dataframe as input and returns the same dataframe with scores added.

**Comment by ahadda5** [comment]
thanks @cmacdonald  , where can I see an example. 
So far all the example(on monoT5 repo) is a pure seq-to-seq where it is still generating, vs sorting/ranking existing labels.

**Comment by ahadda5** [comment]
while the output seq on the custom dataset are great compared to previous models, I 'm only interested in ranking and not generating.

**Comment by cmacdonald** [comment]
I created an example notebook:
https://colab.research.google.com/drive/1rnv1w26k_lZhWT345J1ovMVBBgxLHHRB?usp=sharing

The key code is:
```python

import pandas as pd
from pyterrier_t5 import MonoT5ReRanker
monoT5 = MonoT5ReRanker()

example_input = pd.DataFrame([
    ['q1', 'chemical reactions', 'd1', 'excitation mechanisms of the oxygen emission in the upper atmosphere\nphotochemical reactions affected by mass motions and direct excitation\ndue to mass motions are examined'
],
    ['q1', 'chemical reactions', 'd2', 'ion neutral reactions  a list is given of reactions which may control\nthe chemical composition of positive ions in the ionosphere  under\ncertain assumptions about reaction rate coefficients nitrogen is\nsubstantially dissociated in the f region to the theoretical conclusions\nof other investigators']
    ], columns=['qid', 'query', 'docno', 'text'])

scored = monoT5.transform(example_input)

# ranking is defined by the _descending_ score values, 
# or, equivalently the _ascending_ rank
# NOT by the ordering of the dataframe

# so, we see that d2 is estimated to be more relevant to the query
```

If this kind of approach makes sense, there are lots more such "transformers" available in the PyTerrier ecosystem. See also our tutorials: e.g. https://github.com/terrier-org/searchsolutions2022-tutorial

**Comment by ahadda5** [comment]
I see now how it computes the scores. Thank you for demonstrating. 
However, it is clear this is based on MSMacro corpus.  Maybe I wasn't clear, what if I have my brand new corpus, and I want to train a monoT5(even from a starting checkpoint based off of MSMacro) on my new corpus, then run `.transform `(inference) after that. 
Is that possible with pyterrier?

**Comment by cmacdonald** [comment]
Currently no, this repo is only for /applying/ existing learned models. We find the MSMARCO checkpoint is very good. If you wish for train/fine-tune another checkpoint you would need to use the original monoT5 repo.

**Comment by ahadda5** [comment]
Thanks for clarifying.

**Comment by cmacdonald** [comment]
That's not to say that the monoT5 training code couldnt be ported into this repo, such that it is more easily usable with a more pyterrier-esque API!
